api/src/types/items.ts

```diff
@@ -11,14 +6,10 @@ export type Item = Record<string, any>;
 
 export type PrimaryKey = string | number;
 
-export type Alterations = {
-	create: {
-		[key: string]: any;
-	}[];
-	update: {
-		[key: string]: any;
-	}[];
-	delete: (number | string)[];
+export type Alterations<T extends Item = Item, K extends keyof T | undefined = undefined> = {
+	create: Partial<T>[];
+	update: (K extends keyof T ? Partial<T> & Pick<T, K> : Partial<T>)[];
+	delete: (K extends keyof T ? T[K] : PrimaryKey)[];
```

Comment:
Perhaps im missing something but is the added complexity of `K` required here? I can't currently find an advantage over doing:
```suggestion
export type Alterations<T extends Item = Item> = {
	create: Partial<T>[];
	update: Partial<T>[];
	delete: PrimaryKey[];
```

Reply:
The item props can be partial while the primary key (`K`) should not. This enables us to define the primary key and get the correct type out of the box:
```
Alterations<User, 'id'>
```

Reply:
Ah that makes sense for this implementation specifically as it involves existing users. If that's the case then we should probably name it something more specific as IIRC `someRelation: { create: [{ slug: 'test' }] }` would be valid on the API for "Alterations" syntax